### ISSUE TITLE
Create channel internally

### ISSUE THREAD

**Issue by passcod** [notify-rs/notify#188]
To enable this API instead:

```rust
let mut watcher: RecommendedWatcher = Watcher::new(Duration::from_secs(2))?;
let tx = watcher.channel();
```

Depends on #160

**Comment by passcod** [comment]
Should still have a way to pass a channel in:

```rust
Watcher::with_channel(tx, ...)
```

**Comment by mitchmindtree** [comment]
Closed via #214.
